conans/test/unittests/model/version/test_version_range.py

```diff
@@ -37,7 +37,9 @@
     ['~1.1.2',  [[['>=', '1.1.2-'], ['<', '1.2.0-']]], ["1.1.3"],  ["1.1.3-alpha1", "1.2.0-alpha1"]],
     ['>=2.0-pre.0 <3', [[['>=', '2.0-pre.0'], ['<', '3-']]], ["2.1"], ["2.0-pre.1","2.0-alpha.1"]],
     # Or explicitly
-    ['>=2.0-pre.0, include_prerelease', [[['>=', '2.0-pre.0']]], ["2.1", "2.0-pre.1"], ["1.5"]]
+    ['>=2.0-pre.0, include_prerelease', [[['>=', '2.0-pre.0']]], ["2.1", "2.0-pre.1"], ["1.5"]],
+    # Build metadata
+    ['>=1.0.0+2', [[['>=', '1.0.0+2']]], ["1.0.0+2", "1.0.0+3"], ["1.0.0+1"]],
```

Comment:
We need to add cases of pre-releases for this. Is ``1.0.1-pre.1`` valid if prereleases=True?
What if the condition is ``>=1.0.0-pre.1+2``? Just rough examples, just the idea, not exact cases.